### DISCUSSION TITLE
smaller design?

### DISCUSSION THREAD

**Discussion by grericht** [nliaudat/esp32_8ch_motor_shield#52] [Ideas]
I would like to change the layout a little. I would like to make it a little smaller. I would like to solder some things directly. e.g. the ESP32wroom and possibly also the connections for the servomotors.
I hope it's legitimate to ask: can I get the project file for this? Of course, I would also be happy to work on the project and provide the layout afterwards.
Or have I overlooked the fact that it is already available? Which PCBCad program was used?

**Comment by nliaudat** [comment]
Hi, 
1) Soldering the ESP32 on the board : 
It's definitively not a good idea as you have to reproduce all the ESP32 devkit parts on the pcb. It's very complex and you'll probably do not have success if you are not an electronic engineer. 

2) Smaller design : 
The only way to make it smaller is to solder components on both sides. The price is approx x4.

**Reply by grericht** [reply]
Hello and thanks for the quick reply. Please don't take it as an attack, it's probably a stupid question: Why do all parts of the esp32dev kit have to be rebuilt? I thought the DevKit only did the job of providing USB and getting the chip into programming mode. Both are only necessary for initial programming. This could be done before soldering or via the necessary pins. Then later everything via ota. I would still like to make the pins available on the board. Of course this isn't very comfortable but it would be worth it to me. Which functions still need to be recreated?

I think most people could manage to flash the ESP themselves using the PINs. Most people will have enough experience with this by flashing other devices with Tasmota or ESPHome.

This could create enough space for the board to fit in a 100x100 design and this should make it even cheaper.

For me it would also make sense not to necessarily provide 8 connections for motors. So you could also create designs for 4, 5, 6 or 7. This could also make it more individually usable and cheaper.

I don't want to criticize the project. I think it's fantastic and I have little experience. I'm happy to have found the project and if that doesn't work, then I'll implement it like this. But for me 3 boards with 5 and 1 board with 4 motors would be more suitable. I would also like a permanently soldered (once programmed) ESP32 much better than this plug-in solution.

**Reply by nliaudat** [reply]
Hi, 

- Making a functionnal ESP32 on a custom pbc is not simple as you think. 
Can you calculate the exact lenght, depth and shape of the pcb antenna ?

- #I think most people could manage to flash the ESP themselves using the PINs
Not sure people want to ... and have a JTAG programmer. Choosing an existing ESP32 devkit + esphome was part of the concept to keep it simple for everybody

- The price for soldering + cost of an ESP32 chip is fast the same as a complete devkit. (3.66$ for pbc and 3.8$ for devkit)

- 120x80 is the same pcb cost as 100x100.
The price depends mainly on the SMT assembly.
